src/olympia/promoted/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+
+from olympia.addons.models import Addon
+from olympia.amo.models import ModelBase
+from olympia.constants.promoted import (
+    NOT_PROMOTED, PROMOTED_GROUPS, PROMOTED_GROUPS_BY_ID)
+from olympia.versions.models import Version
+
+
+class PromotedAddon(ModelBase):
+    GROUP_CHOICES = [(group.id, group.name) for group in PROMOTED_GROUPS]
+    group_id = models.SmallIntegerField(
+        choices=GROUP_CHOICES, default=NOT_PROMOTED.id)
+    addon = models.OneToOneField(
+        Addon, on_delete=models.CASCADE, null=False)
+
+    @property
+    def group(self):
+        return PROMOTED_GROUPS_BY_ID.get(self.group_id, NOT_PROMOTED)
+
+    def __getattr__(self, name):
+        # magic to return the fixed properties of the promoted group too
```

Comment:
Is this really necessary ? You could have a `@property` called... `properties` or something that would return them without adding them magically to this class.

Reply:
Its not.  I just added it on a whim because I thought it might be cool.  There already is a `@property` that returns them - `group` above.

Reply:
Yeah I'd remove that, playing with `__getattr__` just for this is asking for trouble later down the road, IMHO. And the `group` `@property` seems enough.

Reply:
https://github.com/mozilla/addons-server/pull/14898/commits/714cae363455aba4f8edf98457e145e8ed18654a